src/hooks/wallets/wallet-connect.ts

```
@@ -0,0 +1,56 @@
+import { Blockchain } from '@dfx.swiss/react/dist/definitions/blockchain';
+import { EthereumProvider } from '@walletconnect/ethereum-provider';
+import { EthereumProvider as EthClient } from '@walletconnect/ethereum-provider/dist/types/EthereumProvider';
+import { useState } from 'react';
+import { AbortError } from '../../util/abort-error';
+import { useBlockchain } from '../blockchain.hook';
+
+export interface WalletConnectInterface {
+  connect: (blockchain: Blockchain) => Promise<string>;
+  signMessage: (msg: string, address: string, blockchain: Blockchain) => Promise<string>;
+}
+
+export function useWalletConnect(): WalletConnectInterface {
+  const [walletConnectClient, setWalletConnectClient] = useState<EthClient>();
+  const { toChainId } = useBlockchain();
+
+  async function connect(blockchain: Blockchain): Promise<string> {
+    const client = walletConnectClient ?? (await setupConnection(blockchain));
+    await client.connect();
+    const result = await client.request<string[]>({ method: 'eth_requestAccounts' });
+    return result[0];
+  }
+
+  async function setupConnection(blockchain: Blockchain): Promise<EthClient> {
+    const chainId = toChainId(blockchain);
+
+    const provider = await EthereumProvider.init({
+      projectId: process.env.REACT_APP_WC_PID ?? '40d5335cbc27edf36ef95389c9d1ac22',
+      chains: [+(chainId ?? 1)],
```

I would throw an error, if `process.env.REACT_APP_WC_PID` is not defined, and remove the fallback value.